src/averageAges.js

```diff
@@ -37,7 +52,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womensOfCenturyArray = people
+    .filter(person => person.sex === FEMALE_SEX)
+    .filter(person => withChildren
+      ? people
+        .some(child => child.mother === person.name)
+      : true);
```

Comment:
Same about 1 filter instead of 2.
This will remove the extra iteration over the array